src/Services/AppInstallActivationHandler.cs

```diff
@@ -51,63 +63,107 @@ protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
 
     protected async override Task HandleInternalAsync(ProtocolActivatedEventArgs args)
     {
-        await AppActivationFlowAsync(args.Uri.Query);
-    }
+        var uri = args.Uri;
+        var parameters = HttpUtility.ParseQueryString(uri.Query);
 
-    private async Task AppActivationFlowAsync(string query)
-    {
-        try
+        if (parameters != null)
         {
-            // Don't interrupt the user if the machine configuration is in progress
-            if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+            var searchQuery = parameters.Get("search");
+            var wingetIdsQuery = parameters.Get("wingetIds");
+
+            if (!string.IsNullOrEmpty(searchQuery))
```

Comment:
I believe this can be reduced to `if (parameters.Get("search") is not null searchQuery)`, ya?  That way the assignment line can be removed.

Reply:
Good call-out. I'll update, thanks.